pulpcore/cli/common/generic.py

```diff
@@ -15,22 +15,218 @@
 )
 
 
+class PulpCommand(click.Command):
+    def get_short_help_str(self, limit: int = 45) -> str:
+        return self.short_help or ""
+
+    def format_help_text(
+        self, ctx: click.Context, formatter: click.formatting.HelpFormatter
+    ) -> None:
+        entity_ctx: PulpEntityContext = ctx.find_object(PulpEntityContext)
+        if self.help is not None:
+            self.help = self.help.format(entity=entity_ctx.ENTITY, entities=entity_ctx.ENTITIES)
+        super().format_help_text(ctx, formatter)
+
+
+class PulpOption(click.Option):
+    def get_help_record(self, ctx: click.Context) -> Tuple[str, str]:
+        synopsis, help_text = super().get_help_record(ctx)
+        entity_ctx: PulpEntityContext = ctx.find_object(PulpEntityContext)
+        help_text = help_text.format(entity=entity_ctx.ENTITY, entities=entity_ctx.ENTITIES)
+        return synopsis, help_text
+
+
+##############################################################################
+# Option callbacks
+
+
+def _href_callback(ctx: click.Context, param: click.Parameter, value: str) -> str:
+    if value is not None:
+        entity_ctx: PulpEntityContext = ctx.find_object(PulpEntityContext)
+        entity_ctx.pulp_href = value
+    return value
+
+
+def _name_callback(ctx: click.Context, param: click.Parameter, value: str) -> str:
+    if value is not None:
+        entity_ctx: PulpEntityContext = ctx.find_object(PulpEntityContext)
+        entity_ctx.entity = {"name": value}
+    return value
+
+
+def _version_callback(ctx: click.Context, param: click.Parameter, value: int) -> int:
+    if value is not None:
+        entity_ctx: PulpEntityContext = ctx.find_object(PulpEntityContext)
+        repository_ctx: PulpRepositoryContext = ctx.find_object(PulpRepositoryContext)
+        entity_ctx.pulp_href = repository_ctx.entity["versions_href"] + str(value)
```

Comment:
How about we allow the version option to be nullable and have the CLI choose the latest version.
```suggestion
def _version_callback(ctx: click.Context, param: click.Parameter, value: int) -> int:
      entity_ctx: PulpEntityContext = ctx.find_object(PulpEntityContext)
      repository_ctx: PulpRepositoryContext = ctx.find_object(PulpRepositoryContext)
      if value is not None:
          entity_ctx.pulp_href = repository_ctx.entity["versions_href"] + str(value)
      else:
          entity_ctx.pulp_href = repository_ctx.entity["latest_version_href"]
```

Reply:
That could work, and spare us the embarrassment of trying to do something without having the version specified.